test/base/EscrowRepayTest.js

```diff
@@ -0,0 +1,74 @@
+// JS Libraries
+const LoansBaseInterfaceEncoder = require("../utils/encoders/LoansBaseInterfaceEncoder");
+const { escrow } = require("../utils/events");
+const { withData } = require("leche");
+const { t } = require("../utils/consts");
+
+// Mock contracts
+const Mock = artifacts.require("./mock/util/Mock.sol");
+const DAI = artifacts.require("./mock/token/DAIMock.sol");
+
+// Smart contracts
+const Escrow = artifacts.require("./mock/base/EscrowMock.sol");
+
+contract("EscrowRepayTest", function(accounts) {
+  const loansEncoder = new LoansBaseInterfaceEncoder(web3);
+
+  let dai;
+  let loans;
+  let instance;
+
+  beforeEach(async () => {
+    dai = await DAI.new();
+
+    loans = await Mock.new();
+    await loans.givenMethodReturnAddress(
+      loansEncoder.encodeLendingToken(),
+      dai.address
+    );
+
+    instance = await Escrow.new();
+    await instance.mockLoans(loans.address);
+  });
+
+  withData({
+    _1_only_owner: [ false, 0, 0, 0, true, "Ownable: caller is not the owner" ],
+    _2_with_escrow_balance: [ true, 1000, 1000, 0, false, null ],
+    _3_with_user_balance: [ true, 1000, 0, 1000, false, null ],
+    _4_with_partial_escrow_balance: [ true, 1000, 800, 200, false, null ],
+    _5_with_partial_escrow_balance_user_no_funds: [ true, 1000, 800, 0, true, "ERC20: transfer amount exceeds balance" ]
+  }, function(
+    isOwner,
+    amount,
+    escrowBalance,
+    userBalance,
+    mustFail,
+    expectedErrorMessage
+  ) {
+    it(t("user", "repay", "Should be able to make a payment on the Escrow's loan.", mustFail), async function() {
+      // Setup
+      const caller = accounts[1];
+      await instance.mockIsOwner(true, isOwner);
+
+      await loans.givenMethodReturn(loansEncoder.encodeRepay(), "0x");
+
+      await dai.mint(instance.address, escrowBalance);
+      await dai.mint(caller, userBalance);
+      await dai.approve(instance.address, escrowBalance - userBalance, { from: caller });
+
+      try {
+        // Invocation
+        await instance.repay(amount, { from: caller });
+
+        const loansRepayCallCount = await loans.invocationCountForMethod.call(loansEncoder.encodeRepay());
+
+        // Assertions
+        assert.equal(loansRepayCallCount.toString(), "1", "Loans.repay was not called.");
+        assert(!mustFail);
```

Comment:
If we decide to emit an event, check that event was emitted.

Reply:
Don't we need to validate any amount after the repayment?

Reply:
If we did emit an event, what should the values be? Emitting an event here and on `LoansBase` seems redundant to me. Only thing I can think that would be useful would be the Escrow's balance of the token? Everything else is already emitting from the LoansBase contract.

Reply:
> If we decide to emit an event, check that event was emitted.

If we did emit an event, what should the values be? Emitting an event here and on LoansBase seems redundant to me. Only thing I can think that would be useful would be the Escrow's balance of the token? Everything else is already emitting from the LoansBase contract.

> Don't we need to validate any amount after the repayment?

There's not much I can do in regards to validating amounts since all I am doing in the `Escrow.repay` function is:
1. Checking the Escrow's and borrower's balance (if the Escrow does not have enough funds, it trys to grab funds from the borrower)
1. Setting an allowance on the Loans contract for the repayment amount
1. Calling `Loans.repay`

So unless I mock the `Loans.repay` to pull funds from the Escrow, theres not much to test.